### ISSUE TITLE
[XLNet] attention_mask / input_mask - Why two `attention_mask` inputs?

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/transformers#7512]
For whatever reason XLNet accepts both an `attention_mask` and a `input_mask`. As far as I understand `attention_mask` = 1 - `input_mask`. I don't think having the same input twice (one is the inverse of the other) has any advantage. We should remove `input_mask` IMO (first depreciate, then remove). Also, `attention_mask` should be put in the 2nd positino to make model compatible with torchscript.

**Comment by LysandreJik** [comment]
I agree! Deprecation until v4.0.0 seems like a reasonable solution,

**Comment by sgugger** [comment]
(We'll need at least one release with it before 4.0.0 if we want to remove the deprecation at 4.0.0)
